### PR TITLE
Core/Utgarde Pinacle: Paralyze Removes On Map Change

### LINKED ISSUE #28777: Core/Utgarde Pinnacle: You get permastunned if you leave instance at Svala sacrifice phase
### Description

You get permastunned if you leave instance at Svala sacrifice phase
You have to relog in order that stun fade out.

https://user-images.githubusercontent.com/24683355/213915421-27b70b69-2f99-4a6c-9f3c-dd7dd9a7c622.mp4

### Expected behaviour

I guess the stunns need to be removed when you are teleported out.

### Steps to reproduce the problem

Go in Utgarde Pinnacle (heroic I tested) then leave party before sacrifice phase 50% pop

### Branch

3.3.5

### TC rev. hash/commit

835e2918319e

### Operating system

Windows 10x64

### Custom changes

None

### PR DESCRIPTION
**Pull request by Alleyaka**
1. Added flags for removing Paralyze from player if they're moved from instance.
2. This fixes issue tracker: https://github.com/TrinityCore/TrinityCore/issues/28777

**Changes proposed:**

**Issues addressed:**

Closes #28777

**Tests performed:**

1. Builds successfully
2. Removes debuff if the player is removed from/leaves the instance during the channel of Paralyze.

**Known issues and TODO list:** (add/remove lines as needed)

### CODE REVIEW

**Comment by Ovahlord** [comment]
hack. The real fix would be properly clearing the aura when the channeling of the casters gets interrupted by teleporting.

**Comment by acidmanifesto** [comment]
> hack. The real fix would be properly clearing the aura when the channeling of the casters gets interrupted by teleporting.

Wrong. The real fix is fixing the whole mess with auras in this core instead of all these spot fixed in the spell fix
 KEK.

**Comment by techett** [comment]
> hack. The real fix would be properly clearing the aura when the channeling of the casters gets interrupted by teleporting.

Thank you for the insight. That was considered; however, this uses flags that are created for this situation.
`spellInfo->AuraInterruptFlags = AURA_INTERRUPT_FLAG_CHANGE_MAP;`

I do agree: there are better ways to accomplish this. This is a fix for the time being until auras can be handled in a more "efficient" manner. Where would you start when attempting to tackle the larger task at hand?

**Comment by Nyr97** [comment]
> hack. The real fix would be properly clearing the aura when the channeling of the casters gets interrupted by teleporting.

Actually, it should be removed if the target gets further than the channeling spells' range instead (which should also work for teleport in that case).

File: src/server/game/Spells/SpellMgr.cpp [hunk 1]
```diff
@@ -3850,6 +3850,12 @@ void SpellMgr::LoadSpellInfoCorrections()
         spellInfo->_GetEffect(EFFECT_0).ApplyAuraName = SPELL_AURA_ADD_PCT_MODIFIER;
     });
 
+    // Utgarde Pinnacle - Paralyze(48278) Removal on change of map
+    ApplySpellFix({ 48278 }, [](SpellInfo* spellInfo)
+    {
+        spellInfo->AuraInterruptFlags = AURA_INTERRUPT_FLAG_CHANGE_MAP;
```
[the remark this answers is not in the document]
Comment (techett) [reply src/server/game/Spells/SpellMgr.cpp]: Updated to include the pipe.

**Review by eXPeRi91 — APPROVED** [review]
Lovely, now I don't see why not to merge this.

**Pull request closed without merging**